fritzconnection/lib/fritztam.py

```diff
@@ -0,0 +1,148 @@
+"""
+# This module is part of the FritzConnection package.
+# https://github.com/kbr/fritzconnection
+
+# License: MIT (https://opensource.org/licenses/MIT)
+# Author: Mark Ullmann
+
+
+from ..core.exceptions import FritzServiceError
+from .fritzbase import AbstractLibraryBase
+from xml.etree import ElementTree as etree
+
+
+# important: don't set an extension number here:
+SERVICE = 'X_AVM-DE_TAM'
+
+
+class FritzTAM(AbstractLibraryBase):
+    """
+    Class to interact with the integrated telephone answering machine (TAM).  All
+    parameters are optional.  If given, they have the following meaning: `fc`
+    is an instance of FritzConnection, `address` the ip of the Fritz!Box,
+    `port` the port to connect to, `user` the username, `password` the
+    password, `timeout` a timeout as floating point number in seconds,
+    `use_tls` a boolean indicating to use TLS (default False). It is
+    recommended to use a dedicated user for security reasons.  Setup of the
+    TAM is not supported, this needs to be done in the web interface.
+    """
+    # This class is adapted from the class FritzWLAN.
+    def __init__(self, *args, service=1, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.service = service
+        self._sid_token = None
+
+    def _action(self, actionname, **kwargs):
+        service = f'{SERVICE}{self.service}'
+        return self.fc.call_action(service, actionname, **kwargs)
+
+    @property
+    def _sid(self):
+        """
+        Authentication token for sid, to access certain urls. This is cached
+        internally.
+        """
+        if self._sid_token is None:
+            sidResponse = self.fc.call_action("DeviceConfig",
+                                              "X_AVM-DE_CreateUrlSID")
+            self._sid_token = sidResponse['NewX_AVM-DE_UrlSID'].split("=")[1]
+        return self._sid_token
+
+    def _get_tam_list_xml(self):
+        """
+        Information about the TAMs. Returns an XML with the data.
+        """
+        result = self._action('GetList')
+        return result['NewTAMList']
+
+    def _get_message_list_xml(self, tamIndex="0"):
+        """
+        Fetches an XML with the list of messages for the specified TAM.
+        """
+        result = self._action("GetMessageList", NewIndex=tamIndex)
+        messageListRequest = self.fc.session.get(result['NewURL'])
+        return messageListRequest.content
+
+    @property
+    def tam_options(self):
+        """
+        Return the general options of all TAM as a dictionary. These values are provided by Fritz!Box.
+
+        Keys: TAMRunning, Stick, Status, Capacity.
+        """
+        root = etree.fromstring(self._get_tam_list_xml())
+        tags = ['TAMRunning', 'Stick', 'Status', 'Capacity']
+        result = {element.tag: element.text for element in root
+                  if element.tag in tags}
+        return result
+
+    # @property
+    def tam_list(self):
```

Comment:
name it `tams` or anything else, but don't but a type into the label (like `_list` here).